pretyping/recordops.ml

```diff
@@ -185,7 +185,8 @@ let rec cs_pattern_of_constr env t =
   | Rel n -> Default_cs, Some n, []
   | Prod (_,a,b) when Vars.noccurn 1 b -> Prod_cs, None, [a; Vars.lift (-1) b]
   | Proj (p, c) ->
-    let { Environ.uj_type = ty } = Typeops.infer env c in
+    let ty = Retyping.get_type_of env (Evd.from_env env) (EConstr.of_constr c) in
+    let ty = EConstr.Unsafe.to_constr ty in
```

Comment:
I would be more keen in adding to Retyping an API like `get_type_of_constr env c` hiding the detour to econstr

Reply:
Done.